modules/dnn/src/layers/attention_layer.cpp

```diff
@@ -24,6 +24,105 @@ static void packWeight(size_t num_heads, size_t head_size, size_t input_hidden_s
     }
 }
 
+
+static void rotationKernel(
+    float* data, const float* rotation_table,
+    size_t seq_len, size_t d
+)
+{
+    CV_Assert(d % 2 == 0);
+    const size_t d_half = d / 2;
+
+    double nstripes = double(seq_len) * d_half * (1.0/1024.0);
+
+    auto fn = [&](const cv::Range& range)
+    {
+        for (int t = range.start; t < range.end; ++t)
+        {
+            float* out_ptr    = data + size_t(t) * d;
+            const float* table_ptr = rotation_table + size_t(t) * d;
+            size_t i = 0;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+            const size_t w = VTraits<v_float32>::vlanes();
+            for (; i + w <= d_half; i += w)
+            {
+                v_float32 sin_v, cos_v, x_even, x_odd;
+                v_load_deinterleave(table_ptr + 2*i, sin_v, cos_v);
+                v_load_deinterleave(out_ptr    + 2*i, x_even, x_odd);
+
+                v_float32 out_even = v_sub(v_mul(cos_v, x_even), v_mul(sin_v, x_odd));
+                v_float32 out_odd  = v_add(v_mul(sin_v, x_even), v_mul(cos_v, x_odd));
+
+                v_store_interleave(out_ptr + 2*i, out_even, out_odd);
+            }
+#endif
+            // scalar tail
+            for (; i < d_half; ++i)
+            {
+                float s  = table_ptr[2*i  ];
+                float c  = table_ptr[2*i+1];
+                float xe = out_ptr[2*i];
+                float xo = out_ptr[2*i+1];
+                out_ptr[2*i]   = xe * c - xo * s;
+                out_ptr[2*i+1] = xo * c + xe * s;
+            }
+        }
+    };
+
+    // This will spin up threads and run fn over [0, seq_len)
+    parallel_for_(cv::Range(0, int(seq_len)), fn, nstripes);
+}
+
+static void precompRotationTable(float *data,
+                                  size_t seq_len,
+                                  size_t d) {
+    // RoPE precomputation
+    // RoPE is a positional encoding method used in transformer models.
+    // It uses sine and cosine functions to encode the position of tokens in a sequence
+    // initially introduced for NLP in https://arxiv.org/pdf/2104.09864
+
+    // assume data is of shape [seq_ken,d]
+    const float  logBase = std::log(10000.0f);
+    const float  inv_d   = 1.0f / float(d);
+    const size_t d_half = d / 2;
+    for (size_t pos = 0; pos < seq_len; ++pos) {
+
+        size_t i = 0;
+        float* data_ptr = data + pos * d;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+        const size_t w = VTraits<v_float32>::vlanes();
+        const v_float32 v_logBase = v_setall_f32(logBase);
+        const v_float32 v_inv_d   = v_setall_f32(inv_d);
+        const v_float32 v_neg2    = v_setall_f32(-2.0f);
+
+        for (; i + w <= d_half; i+=w) {
+            int idx_buf[CV_SIMD_WIDTH];
```

Comment:
CV_SIMD_WIDTH is compile time constant. It may not work correctly with _SCALABLE branch. please use `VTraits<xxx>::max_nlanes` instead. For fixed-size SIMD it works in the same way.

Reply:
@fengyuentau please correct me, if I'm wrong.

Reply:
@asmorkalov,

```
const size_t w = VTraits<v_float32>::vlanes();
...
int idx_buf[w];
```

caused  **[-Wvla-cxx-extension]** in GCC and the build for Win64 failed. 
I wasn't sure what's the proper way to deal with this..

Reply:
Yes, it does not work, because `w` value is not known in compile time. `VTraits<xxx>::max_nlanes` is compile time constant. It's equal to `CV_SIMD_WIDTH` for fixed SIMD size architectures (x86). RISC-V RVV vector size is not known in compile time, but we know maximum vector length and use it for intermediate buffers to fit any feasible vector size.

Reply:
I have replaced  CV_SIMD_WIDTH with VTraits<v_float32>::max_nlanes for both CV_SIMD and CV_SIMD_SCALABLE branch